storm-core/src/jvm/org/apache/storm/trident/Stream.java

```diff
@@ -565,19 +578,169 @@ public Stream aggregate(Fields inputFields, ReducerAggregator agg, Fields functi
                 .aggregate(inputFields, agg, functionFields)
                 .chainEnd();
     }
-    
+
+    /**
+     * Returns a stream of tuples which are aggregated results of a tumbling window with every {@code windowCount} of tuples.
+     *
+     * @param windowCount represents window tuples count
+     * @param inputFields projected fields for aggregator
+     * @param aggregator aggregator to run on the window of tuples to compute the result and emit to the stream.
+     * @param functionFields fields of values to emit with aggregation.
+     *
+     * @return
+     */
+    public Stream tumblingCountWindow(int windowCount, Fields inputFields, Aggregator aggregator, Fields functionFields) {
```

Comment:
The various windowing configurations can be expressed with the `window(WindowConfig windowConfig,...)`. If we are adding wrappers over it, can you please maintain compatibility with the core apis ?

i.e,
```java
tumblingWindow(Count count)
tumblingWindow(Duration duration)
window(Count windowLength, Count slidingInterval)
window(Duration windowLength, Duration slidingInterval)
```